lib/commands/network.js

```diff
@@ -42,6 +45,116 @@ commands.setNetworkConnection = async function (type) {
   return await this.getNetworkConnection();
 };
 
+commands.toggleData = async function () {
+  log.info('Toggling data');
+  let data = !(await this.adb.isDataOn());
+  await this.wrapBootstrapDisconnect(async () => {
+    await this.adb.setWifiAndData({data});
+  });
+};
+
+commands.toggleWiFi = async function () {
+  log.info('Toggling WiFi');
+  let wifi = !(await this.adb.isWifiOn());
+  await this.wrapBootstrapDisconnect(async () => {
+    await this.adb.setWifiAndData({wifi});
+  });
+};
+
+commands.toggleFlightMode = async function () {
+  log.info('Toggling flight mode');
+  let flightMode = !(await this.adb.isAirplaneModeOn());
+  await this.wrapBootstrapDisconnect(async () => {
+    await this.adb.setAirplaneMode(flightMode);
+  });
+  await this.wrapBootstrapDisconnect(async () => {
+    await this.adb.broadcastAirplaneMode(flightMode);
+  });
+};
+
+commands.setGeoLocation = async function (location) {
+  let cmd = `geo fix ${location.longitude} ${location.latitude}`;
```

Comment:
This method only works on emulators (since real devices don't have telnetd running). It might be good to check for read devices and throw a `NotYetImplementedError`, rather than silently failing to connect.